domain/src/test/java/org/oppia/domain/oppialogger/analytics/AnalyticsControllerTest.kt

```diff
@@ -358,13 +399,108 @@ class AnalyticsControllerTest {
     assertThat(eventContext.revisionCardContext.subTopicId).matches(TEST_SUB_TOPIC_ID)
   }
 
+  @ExperimentalCoroutinesApi
+  @InternalCoroutinesApi
+  @Test
+  fun testController_logTransitionEvent_withNoNetwork_checkLogsEventToStore() {
+    networkConnectionUtil.setCurrentConnectionStatus(NONE)
+    analyticsController.logTransitionEvent(
+      TEST_TIMESTAMP,
+      EventAction.EVENT_ACTION_UNSPECIFIED,
+      analyticsController.createQuestionContext(
+        TEST_QUESTION_ID,
+        listOf(
+          TEST_SKILL_LIST_ID, TEST_SKILL_LIST_ID
+        )
+      )
+    )
+
+    val eventLogs = analyticsController.getEventLogs()
+    eventLogs.observeForever(this.mockOppiaEventLogsObserver)
+    testCoroutineDispatchers.advanceUntilIdle()
+    verify(
+      this.mockOppiaEventLogsObserver,
+      atLeastOnce()
+    ).onChanged(oppiaEventLogsResultCaptor.capture())
+
+    val eventLog = oppiaEventLogsResultCaptor.value.getOrThrow().getEventLog(0)
+    assertThat(eventLog.priority).isEqualTo(Priority.ESSENTIAL)
+    assertThat(eventLog.context.activityContextCase).isEqualTo(QUESTION_CONTEXT)
+    assertThat(eventLog.timestamp).isEqualTo(TEST_TIMESTAMP)
+    assertThat(eventLog.actionName).isEqualTo(EventAction.EVENT_ACTION_UNSPECIFIED)
+  }
+
+  @ExperimentalCoroutinesApi
+  @InternalCoroutinesApi
+  @Test
+  fun testController_logClickEvent_withNoNetwork_checkLogsEventToStore() {
+    networkConnectionUtil.setCurrentConnectionStatus(NONE)
+    analyticsController.logClickEvent(
+      TEST_TIMESTAMP,
+      EventAction.EVENT_ACTION_UNSPECIFIED,
+      analyticsController.createQuestionContext(
+        TEST_QUESTION_ID,
+        listOf(
+          TEST_SKILL_LIST_ID, TEST_SKILL_LIST_ID
+        )
+      )
+    )
+
+    val eventLogs = analyticsController.getEventLogs()
+    eventLogs.observeForever(this.mockOppiaEventLogsObserver)
+    testCoroutineDispatchers.advanceUntilIdle()
+    verify(
+      this.mockOppiaEventLogsObserver,
+      atLeastOnce()
+    ).onChanged(oppiaEventLogsResultCaptor.capture())
+
+    val eventLog = oppiaEventLogsResultCaptor.value.getOrThrow().getEventLog(0)
+    assertThat(eventLog.priority).isEqualTo(Priority.OPTIONAL)
+    assertThat(eventLog.context.activityContextCase).isEqualTo(QUESTION_CONTEXT)
+    assertThat(eventLog.timestamp).isEqualTo(TEST_TIMESTAMP)
+    assertThat(eventLog.actionName).isEqualTo(EventAction.EVENT_ACTION_UNSPECIFIED)
+  }
+
+  @ExperimentalCoroutinesApi
+  @InternalCoroutinesApi
+  @Test
+  fun testController_logTransitionEvent_withNoNetwork_exceedLimit_checkEventLogStoreSize() {
+    networkConnectionUtil.setCurrentConnectionStatus(NONE)
+    for (i in 1..5005) {
```

Comment:
Rather than logging a really large number, can we override the limit itself? If you use a Dagger constant for this variable, you can use a test module to limit it to something small (like 2), which then lets you avoid needing a control structure of the test. This in turn simplifies the test, makes it more performant, avoids a dependency on an implementation detail, and makes it easier to verify as correct since there isn't logic in the test that also needs to be verified as correct.

Reply:
Yes, will do.

Reply:
Done.